### DISCUSSION TITLE
Porous Flow-nonlinear solver does not coverge

### DISCUSSION THREAD

**Discussion by rksin8** [idaholab/moose#19034] [Q&A Modules: General] [answered]
Hi,
I am trying to develop a hydromechanical model based on [`Porous Flow THM example`](https://mooseframework.inl.gov/modules/porous_flow/thm_example.htmll). Model runs but non-linear solver does not converge.  [`My Input&Mesh file available here.`](https://github.com/rksin8/moose-TH)

How can I solve this issue?

Thanks

**Comment by GiudGiud** [comment, marked as answer]
Hello

Have you consulted the following pages
https://mooseframework.inl.gov/modules/porous_flow/stabilization.html
https://mooseframework.inl.gov/modules/porous_flow/nonlinear_convergence_problems.html

Guillaume

**Reply by rksin8** [reply]
after scaling, not it is working fine.  Thank you!
